quiz.js

```diff
@@ -0,0 +1,149 @@
+
+            //難易度のdivを作成
+            const divDifficulty = document.createElement('div');
+            divDifficulty.innerHTML = '<b>[難易度] ' + currentQuestion.difficulty + '</b>';
+
+            //質問のdivを作成
+            const divQuestion= document.createElement('div');
+            divQuestion.innerHTML = '<hr>' + currentQuestion.question + '<hr>';
+
+            //回答のdivを作成
+            const divAnswers = document.createElement('div');
+
+            //回答作成順をシャッフル (c = 正解, i0 = 0番目の不正解 etc)
+            const randomOrder = ['c', 'i', 'i', 'i'];
+            shuffle(randomOrder);
+            let indexcount = 0;
+
+            randomOrder.forEach ((value, orderIndex) => {
+                if (value === 'c') {
+                    //正解buttonを作成して、回答divにappend
+                    const correctAnswerBtn = document.createElement('button');
+                    correctAnswerBtn.innerHTML = currentQuestion.correct_answer;
+                    correctAnswerBtn.value = 1;
+                    correctAnswerBtn.id = index;
+                    correctAnswerBtn.addEventListener('click', proceedQuiz);
+                    divAnswers.appendChild(correctAnswerBtn);
+                }
+                else {
+                    //不正解buttonを作成して、回答divにappend
+                    const incorrectAnswerBtn = document.createElement('button');
+                    incorrectAnswerBtn.value = 0;
+                    incorrectAnswerBtn.id = index;
+                    incorrectAnswerBtn.addEventListener('click', proceedQuiz);
+                    incorrectAnswerBtn.innerHTML = currentQuestion.incorrect_answers[indexcount];
+                    divAnswers.appendChild(incorrectAnswerBtn);
+                    indexcount += 1;
+                }
+
+                linebreak = document.createElement('br');
+                divAnswers.appendChild(linebreak);
+            });
+
+            divContainer.appendChild(divHeader);
+            divContainer.appendChild(divCategory);
+            divContainer.appendChild(divDifficulty);
+            divContainer.appendChild(divQuestion);
+            divContainer.appendChild(divAnswers);
+
+            quizContainer.appendChild(divContainer);
+        }
+    );
+}
+
+function shuffle(array) {
+    let currentIndex = array.length, temporaryValue, randomIndex;
+
+    while (0 !== currentIndex) {
+        randomIndex = Math.floor(Math.random() * currentIndex);
+        currentIndex -= 1;
+    
+        temporaryValue = array[currentIndex];
+        array[currentIndex] = array[randomIndex];
+        array[randomIndex] = temporaryValue;
+    }
+    return array;
+}
+
+function readyQuiz(){
+    const loadingSlide = document.getElementById('loadingSlide');
+    loadingSlide.style.display = 'none';       
+
+    const quiz = document.getElementsByClassName('quiz');
+    quiz[0].style.display = '';
+}
+
+function proceedQuiz(e){
+    score += parseInt(e.target.value);
+    quizNumber = parseInt(e.target.id);
```

Comment:
変数 `quizNumber` が宣言されていないようです。
```suggestion
    let quizNumber = parseInt(e.target.id);
```